DSD-509: Chakra HelperErrorText

EdwinGuzman opened the pull request:
Fixes JIRA ticket [DSD-509](https://jira.nypl.org/browse/DSD-509)

## This PR does the following:
- Updates the `HelperErrorText` component with Chakra.

## How has this been tested?

Storybook.

### Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] I have updated the Storybook documentation accordingly.
- [ ] All new and existing tests passed.

### Front End Review:

<!--- This step is done AFTER creating a PR. -->
<!--- Tugboat creates a static Storybook preview URL once the PR is created. -->
<!--- Copy the URL to the relevant Storybook page here. -->

- [ ] View [the example in Storybook]()

EdwinGuzman committed:
init commit converting helperErrorText

EdwinGuzman committed:
Updating the HelperErrorText component to use Chakra and updating all uses of it.

EdwinGuzman committed:
Merging and fixing conflicts from development

bigfishdesign13 commented on src/components/HelperErrorText/HelperErrorText.tsx:
```diff
@@ -28,36 +31,27 @@ export default function HelperErrorText(
       aria-atomic={ariaAtomic}
-      {...attributes}
+      data-isInvalid={isInvalid}
+      aria-live={announceAriaLive ? ariaLive : "off"}
```
We don't use this in many of the other components.  Why is that?

EdwinGuzman replied:
Do you mean the `aria-live` attribute itself? I think it fits best for this type of component where the description test might change (often) and is useful to the user and isn't already being presented (such as through a label or link).

bigfishdesign13 replied:
Got it.  Makes sense.  Thanks.

bigfishdesign13 commented on src/theme/components/helperErrorText.ts:
```diff
@@ -0,0 +1,9 @@
+const helperErrorText = {
+  baseStyle: ({ isInvalid }) => ({
+    marginBottom: "xs",
```
The original component did have `bottom-margin` set to `xs`, but I think this should be removed and the bottom-margin should be 0.  The `Form` component will **now** handle the spacing between form elements.  Once this is set to 0, we will need to review the other components visually to make sure this margin value was not be relied upon.  Off the top of my head, I think `DatePicker` may be the only one that really needs it.

bigfishdesign13 replied:
I had a typo in this comment.  It should have been...
```The Form component will NOW handle the spacing...```

bigfishdesign13 reviewed (changes requested):
The requested changes tie into what I was saying in the PR for DSD-523.

EdwinGuzman committed:
Updating the margin for the HelperErrorText component and the TextInput component

EdwinGuzman commented:
I added the changes from DSD-523 PR #721  in the last commit for this PR. I also updated the `Form` stories so it displays the `DatePicker` component as an example.

EdwinGuzman committed:
Merging from development and updating the changelog

bigfishdesign13 reviewed (approved):
This looks good.  We just need to make sure to fix the spacing issues now present in `DatePicker`.

EdwinGuzman committed:
Merging from development and fixing conflicts
